docs/src/pages/components/data-grid/rendering/rendering.md

```diff
@@ -5,24 +5,262 @@ components: DataGrid, XGrid
 
-## Components
+### Value getter
 
-As part of the customization API, the grid allows to replace and override the following components:
+Sometimes a column might not have a corresponding value and you just want to render a combination of different fields. To do that, you can set the `valueGetter` attribute of `ColDef` as in the example below:
+
+```tsx
+const columns: ColDef[] = [
+  { field: 'id', hide: true },
+  { field: 'firstName', headerName: 'First name', width: 130 },
+  { field: 'lastName', headerName: 'Last name', width: 130 },
+  {
+    field: 'fullName',
+    headerName: 'Full name',
+    width: 160,
+    valueGetter: (params: ValueGetterParams) =>
+      `${params.getValue('firstName') || ''} ${
+        params.getValue('lastName') || ''
+      }`,
+  },
+];
+```
+
+{{"demo": "pages/components/data-grid/rendering/ValueGetterGrid.js", "defaultCodeOpen": false}}
+
+The value generated is used for filtering, sorting, rendering, etc unless overridden by a more specific configuration.
+
+### Value formatter
+
+The value formatters allow you to format values for display as a string.
+For instance, you might want to format a JavaScript date object into a string year.
+
+```tsx
+const columns: ColDef[] = [
+  { field: 'id', hide: true },
+  {
+    field: 'date',
+    headerName: 'Year',
+    valueFormatter: (params: ValueFormatterParams) =>
+      (params.value as Date).getFullYear(),
+  },
+];
+```
+
+{{"demo": "pages/components/data-grid/rendering/ValueFormatterGrid.js", "defaultCodeOpen": false}}
+
+The value generated is used for filtering, sorting, rendering in the cell and outside, etc unless overridden by a more specific configuration.
+
+### Render cell
+
+By default, the grid render the value as a string in the cell.
+It resolves the rendered output in the following order:
+
+1. `renderCell() => ReactElement`
+1. `valueFormatter() => string`
+1. `valueGetter() => string`
+1. `row[field]`
+
+The `renderCell` method of the column definitions is similar to `valueFormatter`.
+However, it trades to be able to only render in a cell in exchange for allowing to return a React node (instead of a string).
+
+```tsx
+const columns: ColDef[] = [
+  { field: 'id', hide: true },
+  {
+    field: 'date',
+    headerName: 'Year',
+    renderCell: (params: ValueFormatterParams) => (
+      <strong>
+        {(params.value as Date).getFullYear()}{' '}
+        <span role="img" aria-label="birthday">
+          🎂
+        </span>
+      </strong>
+    ),
+  },
+];
+```
+
+{{"demo": "pages/components/data-grid/rendering/RenderCellGrid.js", "defaultCodeOpen": false}}
+
+### Render header
+
+You can customize the look of each header with the `renderHeader` method.
+It takes precedence over the `headerName` property.
+
+```tsx
+const columns: ColDef[] = [
+  { field: 'id', hide: true },
+  {
+    field: 'date',
+    width: 500,
```

Comment:
you can add a column type here to render the date formatted

Reply:
Great idea